core/domain/platform_parameter_registry_test.py

```diff
@@ -407,3 +415,12 @@ def test_evaluate_all_parameters(self) -> None:
                 'parameter_b': False,
             }
         )
+
+    def test_all_feature_flags_are_of_bool_type(self) -> None:
+        if constants.EMULATOR_MODE:
```

Comment:
Why do we skip the emulator mode case? There also shouldn't be this sort of if/else logic in tests.

Reply:
I agree. The reason why I'm skipping the emulator mode is because of the way we get the feature-flags. I'm using [get_all_feature_flag_dicts](https://github.com/oppia/oppia/blob/develop/core/domain/platform_feature_services.py#L79) to fetch the data and this function then tries to get the data from the storage layer if not found there then tries to get data from the [parameter_registry](https://github.com/oppia/oppia/blob/develop/core/domain/platform_parameter_registry.py#L53) variable and in emulator mode we do not have data stored and that is why the tests were failing with the error:
`Exception: Platform parameter not found: dummy_feature.`
If it is fine, I can do something like replacing the feature-flags list with some dummy data, initialize them, etc and then add a check, but I'm not sure if that is what we want to do because our aim is to validate the current present feature-flag.

Let me know what do you think.

Reply:
I don't think I follow, sorry. Why do we not have data stored in emulator mode? The code [here](https://github.com/oppia/oppia/blob/develop/core/domain/platform_parameter_list.py) doesn't seem to condition on that.

(And yup, I'm not keen on the solution you mention for the reason you mention.)

Reply:
Yeah, you are right. Please ignore my previous comment, I was thinking something else. I have made the appropriate changes.
The main reason for the error was that the setUp function of the file, It was making changes to the value of parameter_registry` variable.